pkg/loadbalance/interleaved_weighted_round_robin.go

```diff
@@ -0,0 +1,99 @@
+package loadbalance
+
+import (
+	"context"
+	"sync"
+
+	"github.com/bytedance/gopkg/lang/fastrand"
+
+	"github.com/cloudwego/kitex/pkg/discovery"
+)
+
+type iwrrNode struct {
+	discovery.Instance
+	remainder int
+
+	next *iwrrNode
+}
+
+type iwrrQueue struct {
```

Comment:
Since the size is fixed when initializing the Picker, a slice  based queue will be much faster than the pointer based

Reply:
Because nodes are always reused, all operations are just adjusting the pointers, so there is little difference in performance from ring queues